tests/src/test/scala/com/nvidia/spark/rapids/AdaptiveQueryExecSuite.scala

```diff
@@ -90,6 +91,28 @@ class AdaptiveQueryExecSuite
     }
   }
 
+  test("SPARK-30953: InsertAdaptiveSparkPlan should apply AQE on child plan of write commands") {
+
+    val conf = new SparkConf()
+      .set(SQLConf.ADAPTIVE_EXECUTION_ENABLED.key, "true")
+      .set(SQLConf.ADAPTIVE_EXECUTION_FORCE_APPLY.key, "true")
+      .set(RapidsConf.TEST_ALLOWED_NONGPU.key, "DataWritingCommandExec")
+
+      withGpuSparkSession(spark => {
+        frameFromParquet("timestamp-date-test.parquet")(spark)
+          .createOrReplaceTempView("testData")
+
+        val df = spark.sql("CREATE TABLE t1 USING parquet AS SELECT * FROM testData")
+        val plan = df.queryExecution.executedPlan
+        assert(plan.isInstanceOf[DataWritingCommandExec])
```

Comment:
I'm a little unclear what the purpose of this test is.  Normally we should be replacing `DataWritingCommandExec` with `GpuDataWritingCommandExec`, but it looks like this test explicitly expects that to not happen.  Not sure offhand why it's not being replaced -- timezone not UTC and therefore timestamps are problematic maybe?

If whatever's triggering the `DataWritingCommandExec` to not be replaced ultimately gets fixed then this test will have problems.  Might be best to set the config to _force_ it to not be replaced.  But then I question what the real utility of the test is wrt. the plugin.  At that point it seems like we're just verifying that Spark treated a CPU node like it's supposed to, which seems like we're testing Spark rather than testing the plugin.

Reply:
Thanks. After looking at this again, I can see that it was just testing the initial plan and not the final plan because I was missing a call to `df.collect`. I see that we don't replace the `DataWritingCommandExec` in this case because we don't support the `CreateDataSourceTableAsSelectCommand` command, so have added comments about that. I have also added an assertion to check that the child of the write command does get translated to GPU.